packages/flutter/lib/src/widgets/raster_widget.dart

```diff
@@ -0,0 +1,457 @@
+  bool _rasterize;
+  set rasterize(bool value) {
+    if (value == rasterize) {
+      return;
+    }
+    _rasterize = value;
+    notifyListeners();
+  }
+
+  /// Whether the raster widget should use the provided [RasterWidgetFallbackDelegate]
+  /// to render.
+  ///
+  /// Defaults to `false`. Generally, this should only be used if it is known
+  /// that the current platform does not support the raster widget.
+  ///
+  /// Setting this to `true` on a raster widget that does not have a fallback
+  /// delegate will cause an exception to be thrown.
+  bool get fallback => _fallback;
+  bool _fallback;
+  set fallback(bool value) {
+    if (value == fallback) {
+      return;
+    }
+    _fallback = value;
+    notifyListeners();
+  }
+}
+
+/// A widget that replaces its child with a rasterized version of the child.
+///
+/// By default, the child is drawn as is. The default [delegate] simply scales
+/// down the image by the current device pixel ratio and paints it into the
+/// canvas. How this image is drawn can be customized by providing a new
+/// subclass of [RasterWidgetDelegate] to the [delegate] argument.
+///
+/// Caveats:
+///
+/// The contents of platform views cannot be captured by a raster
+/// widget. If a platform view is encountered, then by default an error
+/// is thrown. If [ignorePlatformViews] is true, the rasterization is used
+/// without the platform view included. This may result in an incomplete picture,
+/// but may be a reasonable choice if the platform view is only incidentally included
+/// in a scene. Alternatively, if a [fallback] is provided and a platform view is
+/// detected, then the fallback is used to render the scene instead of the
+/// raster. This is only intended for the rare case when a raster widget
+/// cannot know if a child will have a platform view.
+///
+/// This widget is not supported on the HTML backend of Flutter for the web.
+class RasterWidget extends SingleChildRenderObjectWidget {
+  /// Create a new [RasterWidget].
+  ///
+  /// The [controller] and [child] arguments are required.
+  const RasterWidget({
+    super.key,
+    this.delegate = const _RasterDefaultDelegate(),
+    this.fallback,
+    this.ignorePlatformViews = false,
+    required this.controller,
+    required super.child
+  });
+
+  /// A delegate that allows customization of how the image is painted.
+  ///
+  /// If not provided, defaults to a delegate which paints the child as is.
+  final RasterWidgetDelegate delegate;
+
+  /// The controller that determines when to display the children as an image.
+  final RasterWidgetController controller;
+
+  /// A fallback delegate which is used if the child layers contains a platform view.
+  final RasterWidgetFallbackDelegate? fallback;
+
+  /// Whether the raster widget will ignore platform views in child layers.
+  ///
+  /// Defaults to `false`, which causes an error to be thrown if a platform view
+  /// is encountered.
+  final bool ignorePlatformViews;
```

Comment:
(per in-person conversation this flag is going away so I will not complain about it further in this review)